### PR TITLE
detect-filesize: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4039) ticket:

Describe changes:

- Update detect-filesize unit tests to FAIL/PASS APIs

### CODE REVIEW

**Review by inashivb — APPROVED** [review]
Nice job. LGTM. Thanks! :)

File: src/detect-filesize.c [hunk 1]
```diff
@@ -305,16 +289,17 @@ static int DetectFilesizeSetpTest01(void)
...
         goto cleanup;
 
     if (fsd != NULL) {
-        if (fsd->arg1 == 1 && fsd->arg2 == 3 && fsd->mode == DETECT_UINT_RA)
-            res = 1;
+        FAIL_IF_NOT(fsd->arg1 == 1);
+        FAIL_IF_NOT(fsd->arg2 == 3);
+        FAIL_IF_NOT(fsd->mode == DETECT_UINT_RA);
+
+        PASS;
```
Comment (victorjulien) [inline src/detect-filesize.c:302]: Is it ever valid to have `fsd == NULL`? If not, then the test can be simplified further

Comment (hsadia538) [reply src/detect-filesize.c:302]: `fsd == NULL` leads to cleanup. Should I change it to `FAIL_IF_NULL` ?

Comment (jufajardini) [reply src/detect-filesize.c:302]: I'd say so.

Comment (hsadia538) [reply src/detect-filesize.c:302]: The cleanup steps are happening based on if `fsd==NULL` so should I just add them in sequence?

Comment (jufajardini) [reply src/detect-filesize.c:302]: Clean-up only has to be done in case of success, so you can put them in the very end, with the other clean-up steps.

**Comment by hsadia538** [comment]
Followed by #8087

**Pull request closed without merging**